### ISSUE TITLE
Should we support `!` on nilable class variables?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#16791]
Background: Since its inception, Chapel has supported `if myClass` as a shorthand for `if myClass != nil`, and a recent review of this decision decided to preserve it.

In answering a user question recently, I realized that we don't support `if !myClass` which seems asymmetric to me, and probably more like an oversight than an intentional choice.  This issue asks whether we should support `!` on nilable class variables as a means of saying `myClass == nil`.

**Comment by e-kayrakli** [comment]
> This issue asks whether we should support ! on nilable class variables as a means of saying myClass == nil

Is there any reason we shouldn't support this on non-nilable classes? Obviously `if !myNonNilableClass` will always be false, but I think we should support it for the sake of symmetry and enabling generic programming between nilable and nonnilable types.

Similarly, we support postfix-bang on class variables with nonnilable types (where it is a noop).

**Comment by bradcray** [comment]
> Is there any reason we shouldn't support this on non-nilable classes? Obviously if !myNonNilableClass will always be false, but I think we should support it for the sake of symmetry and enabling generic programming between nilable and nonnilable types.

I agree with that, and didn't mean to suggest we'd only support it for non-nilable classes.  Just wasn't thinking about that case since I was thinking about the case where I'd want to use it most.
